From 6ef6a9828a03024f2635b058b2bcfe472bee863f Mon Sep 17 00:00:00 2001
From: Christian Beier <info@christianbeier.net>
Date: Mon, 20 Jul 2020 22:33:29 +0200
Subject: [PATCH 1/2] libvncserver: don't NULL out internal of the default
 cursor

...otherwise an rfbScreen created after rfbScreenCleanup() was called
gets assigned an invalid cursor struct.

Origin: https://github.com/LibVNC/libvncserver/commit/d138cf90130b
Bug-GNOME: https://gitlab.gnome.org/GNOME/gnome-remote-desktop/-/issues/45
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/libvncserver/+bug/1915410

---
 libvncserver/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libvncserver/main.c b/libvncserver/main.c
index 6477ee8..06efe6e 100644
--- a/libvncserver/main.c
+++ b/libvncserver/main.c
@@ -1021,7 +1021,8 @@ void rfbScreenCleanup(rfbScreenInfoPtr screen)
   FREE_IF(underCursorBuffer);
   TINI_MUTEX(screen->cursorMutex);
 
-  rfbFreeCursor(screen->cursor);
+  if(screen->cursor != &myCursor)
+      rfbFreeCursor(screen->cursor);
 
 #ifdef LIBVNCSERVER_HAVE_LIBZ
   rfbZlibCleanup(screen);
-- 
2.25.1

